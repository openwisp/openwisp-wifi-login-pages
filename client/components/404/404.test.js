import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {shallow} from "enzyme";
import getConfig from "../../utils/get-config";
import loadTranslation from "../../utils/load-translation";
import DoesNotExist from "./404";

jest.mock("../../utils/get-config");
jest.mock("../../utils/load-translation");

const defaultConfig = getConfig("default", true);
const createTestProps = (props) => ({
  orgSlug: "default",
  orgName: "default name",
  page: defaultConfig.components["404_page"],
  setTitle: jest.fn(),
  ...props,
});

describe("<DoesNotExist /> rendering with placeholder translation tags", () => {
  const props = createTestProps();
  it("should render translation placeholder correctly", () => {
    const renderer = new ShallowRenderer();
    const wrapper = renderer.render(<DoesNotExist {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<DoesNotExist /> rendering", () => {
  beforeEach(() => {
    loadTranslation("en", "default");
  });

  it("should render correctly default 404 page without props", () => {
    const renderer = new ShallowRenderer();
    const component = renderer.render(<DoesNotExist />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly custom 404 page with props", () => {
    const props = createTestProps();
    const renderer = new ShallowRenderer();
    const component = renderer.render(<DoesNotExist {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should set title with organisation name", () => {
    const props = createTestProps();
    const wrapper = shallow(<DoesNotExist {...props} />);
    const setTitleMock = wrapper.instance().props.setTitle.mock;
    expect(setTitleMock.calls.pop()).toEqual(["404 Not found", props.orgName]);
  });

  it("should not call setTitle if organization is undefined", () => {
    const props = createTestProps();
    props.page = undefined;
    props.orgName = undefined;
    const wrapper = shallow(<DoesNotExist {...props} />);
    const setTitleMock = wrapper.instance().props.setTitle.mock;
    expect(setTitleMock.calls.length).toBe(0);
  });
});
