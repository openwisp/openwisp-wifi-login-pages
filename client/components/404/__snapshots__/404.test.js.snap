// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DoesNotExist /> rendering should render correctly custom 404 page with props 1`] = `
<div
  className="container content"
  id="not-foud-404"
>
  <div
    className="inner"
  >
    <div
      className="main-column"
    >
      <div
        className="inner"
      >
        <div
          className="row owisp-404-row-1"
        >
          404
        </div>
        <div
          className="row owisp-404-row-2"
        >
          Not Found
        </div>
        <div
          className="row owisp-404-row-3"
        >
          Sorry, we couldn't find the page you were looking for.
        </div>
        <div
          className="row owisp-404-row-4"
        >
          <Link
            className="link"
            to="/default"
          >
            Go back to homepage
          </Link>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DoesNotExist /> rendering should render correctly default 404 page without props 1`] = `
<div
  className="container content"
  id="not-foud-404"
>
  <div
    className="inner"
  >
    <div
      className="main-column"
    >
      <div
        className="inner"
      >
        <div
          className="row owisp-404-row-1"
        >
          Oops!
        </div>
        <div
          className="row owisp-404-row-2"
        >
          404 Not Found
        </div>
        <div
          className="row owisp-404-row-3"
        >
          Sorry, an error has occurred, Requested page not found!
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DoesNotExist /> rendering with placeholder translation tags should render translation placeholder correctly 1`] = `
<div
  className="container content"
  id="not-foud-404"
>
  <div
    className="inner"
  >
    <div
      className="main-column"
    >
      <div
        className="inner"
      >
        <div
          className="row owisp-404-row-1"
        >
          404_H
        </div>
        <div
          className="row owisp-404-row-2"
        >
          404_SUBH
        </div>
        <div
          className="row owisp-404-row-3"
        >
          404_MESSAGE
        </div>
        <div
          className="row owisp-404-row-4"
        >
          <Link
            className="link"
            to="/default"
          >
            HOME_PG_LINK_TXT
          </Link>
        </div>
      </div>
    </div>
  </div>
</div>
`;
